ext/bcmath/tests/str2num_formatting.phpt

```diff
@@ -0,0 +1,45 @@
+--TEST--
+1 and 2 argument of bcadd/bcsub/bcmul/bcdiv/bcmod/bcpowmod/bcpow/bccomp (last one works different then others internally);
+1 argument of bcsqrt - must be well-formed
+--SKIPIF--
+<?php if(!extension_loaded("bcmath")) print "skip"; ?>
+--FILE--
+<?php
+echo bcadd("1", "2"),"\n";
+echo bcadd("1.1", "2", 2),"\n";
+
+echo bcadd("1e1", "2");
+echo bcadd("1,1", "2");
+echo bcadd("Hello", "2");
+echo bcadd("1 1", "2");
+
+echo bccomp("1", "2"),"\n";
+echo bccomp("1.1", "2", 2),"\n";
+
+echo bccomp("1e1", "2");
+echo bccomp("1,1", "2");
+echo bccomp("Hello", "2");
+echo bccomp("1 1", "2");
+?>
```

Comment:
Would be good to test `""` as argument as well.

Reply:
And as a positive case `" 0"` (with a space before) -- I think that should not generate a warning, but with the current implementation probably will.

Reply:
Got it, so I need to make `" 0"` to not generate warning, correct? Not sure if  I can handle it.

Reply:
Yes. Just do like @nikic suggested above, i.e. replace `return str[0] == '0' || str[1] == '\0';` with `return *ptr == '\0'`.